### PR TITLE
Guild member voice control and channel category getter

### PR DESCRIPTION
**Pull request by kubkodev**
Hi,
regarding my email I'm creating PR with some minor changes I've implemented in my fork. I have tested tested voice on my RP server with ~60 playerbase and have investigated no crash. With getting channel category there shouldn't be issue, let me know if you find any issues or you'd like to code something differently, I'll be glad to resolve, thanks.

### CODE REVIEW

**Comment by kubkodev** [comment]
Commits that add function DCC_EncodeString don't have to be merged, I created it just for sending special characters contained in our cp1250 charset (from game).

File: discord-encoding.inc [hunk 1]
```diff
@@ -0,0 +1,171 @@
+stock DCC_EncodeString(const src[], dest[], max_size = sizeof dest)
```
Comment (maddinat0r) [inline discord-encoding.inc]: Please delete your encoding helper include from this PR.

File: src/Channel.cpp [hunk 2]
```diff
@@ -187,10 +215,44 @@ void ChannelManager::Initialize()
...
+			channel->UpdateParentChannel(parent_id);
+		}
+		
+		ChannelId_t const &pawnId = channel->GetPawnId();
+		PawnDispatcher::Get()->Dispatch([pawnId]() mutable
+		{
+			// forward DCC_OnChannelUpdate(DCC_Channel:channel);
+			pawn_cb::Error error;
+			pawn_cb::Callback::CallFirst(error, "DCC_OnChannelUpdate", pawnId);
+		});
```
[the remark this answers is not in the document]
Comment (kubkodev) [reply src/Channel.cpp]: I don't know why I did it like that when everywhere is used same approach. I'll resolve it of course may I have a question what may happen when these functions/value updates aren't placed inside that lambda function?

[the remark this answers is not in the document]
Comment (maddinat0r) [reply src/Channel.cpp]: If the value updates don't happen in the PAWN thread, synchronization issues (possible random crashes, data inconsistency) may occur, as there are no mutexes in place.

**Comment by kubkodev** [comment]
All the issues should be resolved. I'm not sure if I mentioned it before, but I discovered that there was a problem with parsing unsigned values from JSON. I encountered that while I was testing `DCC_GetChannelPosition` what always returned default -1 even though channel was always updated. I got it working in f84fb532984a353ce5e1f2cbe4b68e0b12e5eb33

**Comment by maddinat0r** [comment]
The JSON type check change is fine, it definitely was picky with signed/unsigned types before.
Again, thanks for your pull request!

**Pull request merged**